Controllers/ChessManagerController.cs

```diff
@@ -8,58 +14,87 @@ namespace Chess_Backend.Controllers
+        public ChessManagerController(ILogger<ChessManagerController> logger,
+            MovementFactory movementFactory,
+            IBoardParserService boardParserService,
+            ICompositeValidator compositeValidator,
+            IBoardFactory boardFactory,
+            IBoardHolder boardHolder,
+            IMoveToTilesGenerator compositeMovesGenerator
+            )
         {
             _logger = logger;
-            _gameManager = gameManager;
+            _movementFactory = movementFactory;
+            _boardParserService = boardParserService;
+            _validator = compositeValidator;
+            _boardParserService = boardParserService;
+            this.boardFactory = boardFactory;
+            this.boardHolder = boardHolder;
+            this.compositeMovesGenerator = compositeMovesGenerator;
         }
 
-        [HttpPost("StartNewGame")]
-        public ActionResult<string> StartNewGame()
-        {
-            _gameManager.StartNewGame();
-            _logger.LogInformation("Started a new chess game.");
-            return Ok(_gameManager.CurrentFen);
-        }
 
 
-        [HttpGet("GetCurrentFen")]
-        public ActionResult<string> GetCurrentFen()
+        [HttpGet("GetInitialFen")]
+        public IActionResult GetInitialFen()
         {
-            _logger.LogInformation("Retrieving current FEN.");
-            return Ok(_gameManager.CurrentFen);
-        }
+            try
+            {
+                boardHolder.SetBoard(boardFactory.InitializeNewBoard());
+                var fen = _boardParserService.BoardToFen(boardHolder.GetBoard());
+                return Ok(fen); 
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Failed to generate FEN: {Exception}", ex);
+                return StatusCode(500, new { Message = "Internal Server Error. Please check the logs for more details." });
 
+            }
+        }
 
-        [HttpPost("ValidateMove")]
-        public ActionResult<string> ValidateMove([FromBody] MoveRequest move)
+        [HttpPost("move")]
+        public IActionResult MakeMove([FromBody] MoveRequest request)
         {
             try
             {
-                if (_gameManager.TryMakeMove(move.From, move.To, out string newFen))
+                var fromTile = ChessNotationConverter.ConvertToTile(request.From);
+                var toTile = ChessNotationConverter.ConvertToTile(request.To);
+                IBoard currentBoard = boardHolder.GetBoard();
+                var movement = _movementFactory.CreateMovement(fromTile, toTile, currentBoard);
+                if (_validator.IsMovementValid(movement, currentBoard))
                 {
-                    return Ok(newFen);
+                    Piece piece = currentBoard.GetPieceByTilePosition(fromTile)!;
+                    var possibleMoves = compositeMovesGenerator.GetPossibleMoves(piece);
+                    var validMove = possibleMoves.Any(tile => tile.Equals(toTile));
```

Comment:
you have the validator for a reason, you shouldn't validate it at the contorller level

Reply:
why do you validate twice?